### PR TITLE
Add min_evals parameter to EarlyStopping handler

### LINKED ISSUE #3411: Enhancements to EarlyStopping handler
## 🚀 Feature

I would propose several enhancements to the  [EarlyStopping handler](https://docs.pytorch.org/ignite/_modules/ignite/handlers/early_stopping.html#EarlyStopping).

1. Introduce a new parameter, `min_delta_mode`, mimicking the parameter `threshold_mode` of [ReduceLROnPlateau](https://docs.pytorch.org/docs/stable/generated/torch.optim.lr_scheduler.ReduceLROnPlateau.html). If `min_delta_mode="abs"` (default) everything works as of now, while if `min_delta_mode="rel"` the score comparison becomes `score >= best_score (1 + min_delta)`.
2. Introduce a new parameter `mode` that work's as in [ReduceLROnPlateau](https://docs.pytorch.org/docs/stable/generated/torch.optim.lr_scheduler.ReduceLROnPlateau.html), allowing to handle validation metrics to be minimized.

Other ideas I'd like to discuss are:

-  Natively implementing `min_epochs`/`min_evals`, preventing EarlyStopping from terminating until it has been evaluated at least X times. I know this could be addressed as in #3317 but this solution might be more compact? 
- It would be nice to have the same `threshold` and `threshold_mode` nomenclature as `ReduceLROnPlateau`, but I realize this would probably break older code.
- I personally find the parameter name `cumulative_delta` quite counter-intuitive. When I first read about that parameter, I understood it was taking the cumulative sum of the improvements and comparing this cumulative sum to `min_delta`.

### PR DESCRIPTION
**Pull request by D0-T**
Closes #3411

Description:

Adds a 'min_evals' parameter to 'EarlyStopping' to implement a warmup period.
When specified, the handler tracks the best score but does not enforce patience
until it has been called at least 'min_evals' times.
Changes:

- 'ignite/handlers/early_stopping.py': new 'min_evals' param, updated '__call__', 'reset', 'state_dict', 'load_state_dict'

Check list:

- [x] New tests are added (if a new feature is added)
- [x] New doc strings: description and/or example code are in RST format
- [ ] Documentation is updated (if required)

### CODE REVIEW

**Comment by aaishwarymishra** [comment]
@vfdev-5 What do you think? like pr has some problems but should we add helper function like discussed in the issue or add it as parameter so I can ask for refactoring?

**Comment by vfdev-5** [comment]
I think let's finish first with https://github.com/pytorch/ignite/pull/3619
Personally, I would prefer we add a helper function like discussed in the issue instead of a new arg.
